### ISSUE TITLE
Bolt cookie in frontend

### ISSUE THREAD

**Issue by miguelavaqrod** [bolt/bolt#3413]
Hi!

I noticed that now, in frontend, bolt session cookie is not set.
Maybe it can be ok for most cases, but I feel that you are not taking into account the use cases that involve extensions.
There are, and there will be, few extensions that make use of that cookie to store relevant info needed to show in frontend (carts, frontend areas that will be only visible to logged in users, and so on...).

Better than just removing it, I think that it should be a choice developers have to select (maybe just in the main config file).

For now, I had to modify Application.php file by hand to avoid that behaviour.

What do you think about?

Thanks...

**Comment by GawainLynch** [comment]
This [example might help](https://github.com/GawainLynch/bolt-extension-clientlogin/blob/master/src/Controller/ClientLoginController.php#L73-L87)

Bolt's backend cookie shouldn't be relied upon frontend.

**Comment by miguelavaqrod** [comment]
Hi GawainLynch!

I see... Ok then.. no comments about ;-)
If you need a cookie... just create it jejeje

Thanks for your quick response.

**Comment by miguelavaqrod** [comment]
Just a small comment...

I think several extensions are not going to work anymore if they are not updated (at least mines ;-)

**Comment by GawainLynch** [comment]
Hehe… I was literally working on that this morning for another problem.  We good to close this?

**Comment by miguelavaqrod** [comment]
Closing.... ;-)

**Comment by GawainLynch** [comment]
Yeah, it's a double edged sword. The cookies can interfere with caching like Varnish.

**Comment by miguelavaqrod** [comment]
True! I saw the previous issue about that.

**Comment by miguelavaqrod** [comment]
Sorry GawainLynch.. but this maybe has something to be with another problem I suffered this morning...

It was impossible for me to send POST forms from the frontend... it always complained about bad CSFR token.
Refreshing the page, always gave me the same token, and just the first time it was valid.

Maybe has it something to be with not setting the cookie in frontend?

**Comment by GawainLynch** [comment]
How are you POSTing and with what? The token is set in the the HTML form, eg:

```html
<input type="hidden" id="submission__token" name="submission[_token]" value="317aee8dab9b93a3505c749cc9ea1214757b5f65">
```

Have a look at how BoltForms handles it, that's used pretty heavily to do this quite successfully.

**Comment by miguelavaqrod** [comment]
First I just use the form created.
In twig: {{ form_widget(form) }}

In extension:
---
        $form = $this->app['form.factory']->createBuilder('form', $data)
                     ->add('username', 'text')
                     ->add('password', 'password')
                     ->getForm();
....
So token is set.

But when sending it complained about CSRF mismatch.

Then I tried to add this to the form:
<input type="hidden" name="bolt_csrf_token" id="bolt_csrf_token" value="{{ token() }}" />

Desesperate try... I know.. but knowledge about this topic is small from my part...

With both.. no success too :-(

**Comment by GawainLynch** [comment]
@miguelavaqrod have [a look here](https://github.com/GawainLynch/bolt-extension-boltbb/blob/master/src/Controller/BoltBBController.php#L244-L269), I think this is the most straightforward example I can point to.

BoltForms is slightly more complicated to read as things are in a few places.

**Comment by miguelavaqrod** [comment]
OK.. I will study it ;-)

Thanks a lot!!

**Comment by miguelavaqrod** [comment]
Nice! Let the developer decide... ;-)

**Comment by GawainLynch** [comment]
Yeah, we're going to address both needs properly in 2.3… but for now this will restore default behaviour.

**Comment by miguelavaqrod** [comment]
Hi!

I think config.yml.dist should be upgraded to...

**Comment by GawainLynch** [comment]
There is no need. The setting that we've added is temporary for anyone that needs to actually enable the behaviour, and it looks like enabling the behaviour is breaking more than intended anyway.

What #3427 does is restores what has happened normally before #3309. When we branch `release/2.2` and life the feature freeze, we're going to see if we can't find a better way of achieving what was required in #3309… but for now the rest of us are safe :grin:

**Comment by miguelavaqrod** [comment]
;-) OK!! Let's hope you find a all-in-one solution! (so we can stop writing in a closed issue hehehe)

Thanks!

**Comment by bobdenotter** [comment]
Fixed by #3564
